vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListImagesController.java

```diff
@@ -38,12 +47,20 @@ public String listSpaces(@PathVariable(required = false) String page,
         model.addAttribute("totalPages", imageService.getTotalPages());
         model.addAttribute("currentPageNumber", pageNo);
         model.addAttribute("totalImageCount", imageService.getTotalImageCount());
-        model.addAttribute("images", imageService.getImages(pageNo, sortedBy, order));
+        List<IVSImage> images = imageService.getImages(pageNo, sortedBy, order);
+        LinkedHashMap<String, List<ISpace>> imageToSpaces = new LinkedHashMap<String, List<ISpace>>();
+        for(IVSImage image : images) {
+            List<ISpace> spaces = spaceManager.getSpacesWithImageId(image.getId());
```

Comment:
What type of exceptions can spaceManager.getSpacesWithImageId raise?
image Id null or invalid image id?
How is it handled?

Reply:
It is a referenced primary key and hence cannot take null value. Handled on database table configuration.

Reply:
But what about invalid image id?